src/platform/unix.rs

```diff
@@ -56,7 +56,7 @@ pub type XlibErrorHook =
 pub fn register_xlib_error_hook(hook: XlibErrorHook) {
     // Append new hook.
     unsafe {
-        XLIB_ERROR_HOOKS.lock().push(hook);
+        XLIB_ERROR_HOOKS.lock().unwrap().push(hook);
```

Comment:
While this is entirely fine, I'd suggest having a macro like this somewhere in the crate:

```rust
macro_rules! lock {
    ($mutex: expr) => {{
        match ($mutex).lock() {
            Ok(guard) => guard,
            Err(err) => {
                log::warn!("Mutex is poisoned: {:?}", &err);
                err.into_inner()
            }
        }
    }}
}
```

And instead calling it like this:

```rust
lock!(XLIB_ERROR_HOOKS).push(hook);
```

This ensures two things:

1. A poisoned lock won't blow up the entire system.
2. If we decide to switch to another mutex scheme, for some reason, we can easily adjust `lock!` to fit our use case.

Reply:
I went with `unwrap` because it's what's used in the other backends, and I'm not at all sure how to handle mutex poisoning

Reply:
Hmm, I guess I should add it in a separate PR then...

Reply:
That would be nice, please do!

Reply:
Great! I'll wait until after this PR is merged, so I can make a macro that works across the entire crate.

Reply:
Hey @notgull, this PR has been merged, so if you want to make this change, now you have the opportunity!